### PR TITLE
add spot training examples

### PR DESCRIPTION
**Pull request by ishaaq**
*Description of changes:*
Sample notebooks for model training with SageMaker Managed Spot Training.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sagemaker-python-sdk/spot_training_mxnet/spot_training_mxnet.ipynb [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    "custom_code_upload_location = 's3://{}/customcode/mxnet'.format(bucket)\n",
+    "\n",
+    "# Location where results of model training are saved.\n",
+    "model_artifacts_location = 's3://{}/artifacts'.format(bucket)\n",
+    "\n",
+    "# IAM execution role that gives SageMaker access to resources in your AWS account.\n",
+    "# We can use the SageMaker Python SDK to get the role from our notebook environment. \n",
+    "role = get_execution_role()\n",
+    "\n",
+    "import boto3\n",
```
Comment (apacker) [inline sagemaker-python-sdk/spot_training_mxnet/spot_training_mxnet.ipynb]: Minor: move this import and the train/test data location definitions down to the cell where it gets used (above where the MXNet estimator is instantiated)

File: sagemaker-python-sdk/spot_training_tensorflow_estimator/spot_training_tensorflow_estimator.ipynb [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    "        x={INPUT_TENSOR_NAME: np.array(training_set.data)},\n",
+    "        y=np.array(training_set.target),\n",
+    "        num_epochs=None,\n",
+    "        shuffle=True)()\n",
+    "\n",
+    "def serving_input_fn(hyperparameters):\n",
+    "    feature_spec = {INPUT_TENSOR_NAME: tf.FixedLenFeature(dtype=tf.float32, shape=[4])}\n",
+    "    return tf.estimator.export.build_parsing_serving_input_receiver_fn(feature_spec)()\n",
+    "\n",
+    "import boto3\n",
```
Comment (apacker) [inline sagemaker-python-sdk/spot_training_tensorflow_estimator/spot_training_tensorflow_estimator.ipynb]: Minor: move this down to the cell where region gets used

File: sagemaker-python-sdk/spot_training_mxnet/spot_training_mxnet.ipynb (hunk 1 above)
Comment (ishaaq) [reply sagemaker-python-sdk/spot_training_mxnet/spot_training_mxnet.ipynb]: Normally I'd agree, but these new example notebooks are primarily to highlight what you'd need to do enable Managed Spot Training, so I wanted to pare down that last cell to just the spot-specific changes. The rest of the code is simply copied verbatim from the original MXNet notebook and there's nothing remarkable/new about it.

File: sagemaker-python-sdk/spot_training_tensorflow_estimator/spot_training_tensorflow_estimator.ipynb (hunk 2 above)
Comment (ishaaq) [reply sagemaker-python-sdk/spot_training_tensorflow_estimator/spot_training_tensorflow_estimator.ipynb]: see answer to previous comment

**Pull request merged**